pandas/core/indexes/datetimes.py

```diff
@@ -366,9 +366,11 @@ def __new__(cls, data=None,
                 pass
 
         if data is None:
+            values_present = kwargs.pop('values_present', False)
```

Comment:
what the heck is all of this?

Reply:
sorry, what do you mean?

Reply:
that is a change  to fix a problem with resampling for tz-aware series in one of the edge cases
I hit this problem often unfortunately and from my understanding of the design I created a solution that doesn't break other functionality but fixes the problem I have
sorry if the solution doesn't look elegant, if you think there is other one that we can use, please let me know

Reply:
I am not really sure what you are doing. Your changes should be restriced to pandas/core/resample.py

Reply:
if you mean that I should not change anything in pandas/core/resample.py then unfortunately I don't know how to avoid that

Reply:
I mean the opposite. There's where the change should be.

Reply:
I can implement a solution with changes in `pandas/core/resample.py` only I think
let me try

Reply:
yeah I think this is a specific change that resample needs, not generically for date_ranges.